test/OpenIddict.Abstractions.Tests/Primitives/OpenIddictExtensionsTests.cs

```diff
@@ -0,0 +1,872 @@
+            };
+
+            // Act and assert
+            Assert.Equal(result, request.HasScope(OpenIddictConstants.Scopes.OpenId));
+        }
+
+        [Fact]
+        public void IsNoneFlow_ThrowsAnExceptionForNullRequest()
+        {
+            // Arrange
+            var request = (OpenIddictRequest) null;
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() => request.IsNoneFlow());
+
+            Assert.Equal("request", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData(null, false)]
+        [InlineData("unknown", false)]
+        [InlineData("none", true)]
+        [InlineData("none ", true)]
+        [InlineData(" none", true)]
+        [InlineData("none id_token", false)]
+        [InlineData(" none id_token", false)]
+        [InlineData("none id_token ", false)]
+        [InlineData(" none id_token ", false)]
+        [InlineData("NONE", false)]
+        [InlineData("NONE ", false)]
+        [InlineData(" NONE", false)]
+        [InlineData("NONE ID_TOKEN", false)]
+        [InlineData(" NONE ID_TOKEN", false)]
+        [InlineData("NONE ID_TOKEN ", false)]
+        [InlineData(" NONE ID_TOKEN ", false)]
+        public void IsNoneFlow_ReturnsExpectedResult(string type, bool result)
+        {
+            // Arrange
+            var request = new OpenIddictRequest
+            {
+                ResponseType = type
+            };
+
+            // Act and assert
+            Assert.Equal(result, request.IsNoneFlow());
+        }
+
+        [Fact]
+        public void IsAuthorizationCodeFlow_ThrowsAnExceptionForNullRequest()
+        {
+            // Arrange
+            var request = (OpenIddictRequest) null;
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() => request.IsAuthorizationCodeFlow());
+
+            Assert.Equal("request", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData(null, false)]
+        [InlineData("unknown", false)]
+        [InlineData("code", true)]
+        [InlineData("code ", true)]
+        [InlineData(" code", true)]
+        [InlineData("code id_token", false)]
+        [InlineData(" code id_token", false)]
+        [InlineData("code id_token ", false)]
+        [InlineData(" code id_token ", false)]
+        [InlineData("CODE", false)]
+        [InlineData("CODE ", false)]
+        [InlineData(" CODE", false)]
+        [InlineData("CODE ID_TOKEN", false)]
+        [InlineData(" CODE ID_TOKEN", false)]
+        [InlineData("CODE ID_TOKEN ", false)]
+        [InlineData(" CODE ID_TOKEN ", false)]
+        public void IsAuthorizationCodeFlow_ReturnsExpectedResult(string type, bool result)
+        {
+            // Arrange
+            var request = new OpenIddictRequest
+            {
+                ResponseType = type
+            };
+
+            // Act and assert
+            Assert.Equal(result, request.IsAuthorizationCodeFlow());
+        }
+
```

Comment:
Not necessarily for this PR, but we'll need new tests for the extensions introduced as part of the device flow support (mostly a not for myself, but let me know if you're interested in doing it)

Reply:
If it's small changes I can add them here, just let me know what methods to cover, otherwise it may be better to do in separate PR

Reply:
Only 1 extension and 4 properties were needed for the device flow, so it should be trivial:

https://github.com/openiddict/openiddict-core/blob/af1fb2e8907b3dce6d50558a6b90874a6680837b/src/OpenIddict.Abstractions/Primitives/OpenIddictExtensions.cs#L474-L488

https://github.com/openiddict/openiddict-core/blob/af1fb2e8907b3dce6d50558a6b90874a6680837b/src/OpenIddict.Abstractions/Primitives/OpenIddictRequest.cs#L195-L202

https://github.com/openiddict/openiddict-core/blob/af1fb2e8907b3dce6d50558a6b90874a6680837b/src/OpenIddict.Abstractions/Primitives/OpenIddictRequest.cs#L420-L427

https://github.com/openiddict/openiddict-core/blob/af1fb2e8907b3dce6d50558a6b90874a6680837b/src/OpenIddict.Abstractions/Primitives/OpenIddictResponse.cs#L87-L94

https://github.com/openiddict/openiddict-core/blob/af1fb2e8907b3dce6d50558a6b90874a6680837b/src/OpenIddict.Abstractions/Primitives/OpenIddictResponse.cs#L177-L184

Reply:
Ok, I will cover this

Reply:
Done